src/command/verify/verify.c

```diff
@@ -0,0 +1,1124 @@
+                infoBackupMove(result, memContextPrior());
+            }
+        }
+    }
+    MEM_CONTEXT_TEMP_END();
+
+    FUNCTION_LOG_RETURN(INFO_BACKUP, result);
+}
+
+/***********************************************************************************************************************************
+Check the history in the info files
+***********************************************************************************************************************************/
+void
+verifyPgHistory(const InfoPg *archiveInfoPg, const InfoPg *backupInfoPg)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(INFO_PG, archiveInfoPg);                // Postgres information from the archive.info file
+        FUNCTION_TEST_PARAM(INFO_PG, backupInfoPg);                 // Postgres information from the backup.info file
+    FUNCTION_TEST_END();
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        // Check archive.info and backup.info current PG data matches. If there is a mismatch, verify cannot continue since
+        // the database is not considered accessible during the verify command so no way to tell which would be valid.
+        InfoPgData archiveInfoPgData = infoPgData(archiveInfoPg, infoPgDataCurrentId(archiveInfoPg));
+        InfoPgData backupInfoPgData = infoPgData(backupInfoPg, infoPgDataCurrentId(backupInfoPg));
+        checkStanzaInfo(&archiveInfoPgData, &backupInfoPgData);
+
+        unsigned int archiveInfoHistoryTotal = infoPgDataTotal(archiveInfoPg);
+        unsigned int backupInfoHistoryTotal = infoPgDataTotal(backupInfoPg);
+
+        String *errMsg = strNew("archive and backup history lists do not match");
+
+        if (archiveInfoHistoryTotal != backupInfoHistoryTotal)
+            THROW(FormatError, strZ(errMsg));
+
+        // Confirm the lists are the same
+        for (unsigned int infoPgIdx = 0; infoPgIdx < archiveInfoHistoryTotal; infoPgIdx++)
+        {
+            InfoPgData archiveInfoPgHistory = infoPgData(archiveInfoPg, infoPgIdx);
+            InfoPgData backupInfoPgHistory = infoPgData(backupInfoPg, infoPgIdx);
+
+            if (archiveInfoPgHistory.id != backupInfoPgHistory.id ||
+                archiveInfoPgHistory.systemId != backupInfoPgHistory.systemId ||
+                archiveInfoPgHistory.version != backupInfoPgHistory.version)
+            {
+                THROW(FormatError, strZ(errMsg));
+            }
+        }
+    }
+    MEM_CONTEXT_TEMP_END();
+
+    FUNCTION_TEST_RETURN_VOID();
+}
+
+/***********************************************************************************************************************************
+Populate the WAL ranges from the provided, sorted, WAL files list for a given archiveId
+***********************************************************************************************************************************/
+static void
+createArchiveIdRange(ArchiveResult *archiveIdResult, StringList *walFileList, unsigned int *jobErrorTotal)
```

Comment:
General comment on function naming. It's better if all the functions start with `verify` to avoid namespace conflicts with other modules when they are pulled in for unit testing.

Reply:
fixed: eea3f4ebc8bd618cf23ca6b460a1daaf5083a4e5